packages/flutter/lib/src/widgets/implicit_animations.dart

```diff
@@ -1003,18 +1005,20 @@ class AnimatedOpacity extends ImplicitlyAnimatedWidget {
   }
 }
 
-class _AnimatedOpacityState extends AnimatedWidgetBaseState<AnimatedOpacity> {
+class _AnimatedOpacityState extends ImplicitlyAnimatedWidgetState<AnimatedOpacity> {
   Tween<double> _opacity;
 
   @override
   void forEachTween(TweenVisitor<dynamic> visitor) {
-    _opacity = visitor(_opacity, widget.opacity, (dynamic value) => new Tween<double>(begin: value));
+    _opacity = visitor(_opacity, widget.opacity, (dynamic value) {
+      return new Tween<double>(begin: value);
+    });
   }
 
   @override
   Widget build(BuildContext context) {
-    return new Opacity(
-      opacity: _opacity.evaluate(animation),
+    return new FadeTransition(
+      opacity: _opacity.animate(controller),
```

Comment:
Won't this be creating a new animation every build?

Reply:
Yeah.  I moved the .animate() call into forEachTween(), which is where new _opacity Tweens are created.  I think we have to create a new animation every time we create a new Tween, right?

Functionally speaking, I think the forEachTween() method will end up being called almost as often as build(), so I'm not sure it really solves the re-creation problem, but I'm not sure how we can get more efficient than that.  Thoughts?